packages/plugin-ext/src/plugin/env-variables-server-ext.ts

```diff
@@ -0,0 +1,58 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { RPCProtocol } from '../common/rpc-protocol';
+import { PLUGIN_RPC_CONTEXT, EnvMain } from '../common';
+import { EnvVariablesServer, EnvVariable } from '@theia/core/lib/common/env-variables';
+
+export class EnvVariablesServerExt implements EnvVariablesServer {
```

Comment:
Why does in plugin code needs to implement theia core apI? I see that many new methods are not used to implement theia.d.ts. Please remove it does not help supporting VS Code APIs.

Reply:
We've added some new methods into Theia APIs over VSCode APIs. They were required for some Theia plugins features. Does having extra APIs hurt?
As about my changes. Yes, I added new methods which are not used in `theia.d.ts` implementation, but they are needed to comply with `EnvVariablesServer` interface. I believe, that refactoring of this is out of scope for this PR.

Reply:
We have conventions that main and ext shape should be defined in plugin api protocol file, here we don't follow it but trying to reuse code from core which can pull core deps in plugin host process, in additional we add methods which are not needed. cc @benoitf @tsmaeder

Reply:
@akosyakov I completely agree with you. What I am telling is that such refactoring is out of scope of this PR. I prefer to have a PR which fixes one specific issue. Otherwise we will have a mess in PR, long terms PRs and the process of contributing is getting too long.

Reply:
I think such refactoring makes the PR smaller, since you remove a lot of code instead of adding new code trying to save the bogus approach.